flake.nix

```diff
@@ -6,27 +6,96 @@
     nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
   };
 
-  outputs = { self, nixpkgs, utils }:
+  outputs =
+    {
+      self,
+      nixpkgs,
+      utils,
+    }:
     let
-      pkgsForSystem = system: import nixpkgs {
-        inherit system;
-      };
-    in utils.lib.eachSystem [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ] (system: rec {
-      legacyPackages = pkgsForSystem system;
-      devShells.default = with legacyPackages; mkShell {
-        nativeBuildInputs = [
-          cargo
-          clippy
-          pkg-config
-          rustc
-          rustfmt
-          elmPackages.elm
-          elmPackages.elm-format
-        ];
+      pkgsForSystem =
+        system:
+        import nixpkgs {
+          inherit system;
+        };
+    in
+    utils.lib.eachSystem [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ] (
+      system:
+      let
+        pkgs = pkgsForSystem system;
+        web-up =
+          let
+            watchexecCmd = ''
+              watchexec \
+                --shell=none \
+                --workdir=./frontend \
+                --watch=./src \
+                --exts=elm \
+                -- \
+                elm make ./src/Main.elm \
+                --output ./static/main.js
+            '';
+            liveserverCmd = ''
+              live-server \
+                --host=127.0.0.1 \
+                --port=35249 \
+                --hard \
+                ./frontend/static
+            '';
+            cargoCmd = ''
+              cargo \
+                run \
+                --manifest-path=./backend/Cargo.toml \
+                --package=eka_ci_server \
+                --features=bundle-proxy \
+                -- \
+                --bundle-port=35429
+            '';
+          in
+          pkgs.writeShellApplication {
+            name = "web-up";
+            runtimeInputs = [
+              pkgs.mktemp
+              pkgs.mprocs
+              pkgs.watchexec
+              pkgs.live-server
+              pkgs.stdenv.cc
+              pkgs.cargo
+              pkgs.rustc
+            ];
+            text = ''
+              XDG_RUNTIME_DIR=$(mktemp -d)
+              export XDG_RUNTIME_DIR
+              mprocs \
+                --names=cargo,watchexec,live-server \
+                "${cargoCmd}" "${watchexecCmd}" "${liveserverCmd}"
+            '';
+          };
```

Comment:
I think that we now have two package expressions in the flake, let's move them to their own folder, I'll move the dev shell, if you could move this in this PR :)

Reply:
Can you maybe move the dev shell and then I can use that as example? Not super confident in how that should look like